src/main/java/com/mparticle/kits/IterableKit.java

```diff
@@ -0,0 +1,86 @@
+package com.mparticle.kits;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.Bundle;
+
+import com.mparticle.DeepLinkResult;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ *
+ * This is an mParticle kit, used to extend the functionality of mParticle SDK. Most Kits are wrappers/adapters
+ * to a 3rd party SDK, primarily used to map analogous public mParticle APIs onto a 3rd-party API/platform.
+ */
+public class IterableKit extends KitIntegration implements KitIntegration.ActivityListener {
+
+   private String deeplinkUrl;
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return "Iterable";
+    }
+
+    @Override
+    public List<ReportingMessage> setOptOut(boolean optedOut) {
+        ReportingMessage optOutMessage = new ReportingMessage(this, ReportingMessage.MessageType.OPT_OUT, System.currentTimeMillis(), null);
+        return null;
+    }
+
+    @Override
+    public void checkForDeepLink() {
+        IterableHelper.IterableActionHandler clickCallback = new IterableHelper.IterableActionHandler(){
+            @Override
+            public void execute(String result) {
+                DeepLinkResult deepLinkResult = new DeepLinkResult().setLink(result);
+                getKitManager().onResult(deepLinkResult);
```

Comment:
Reading through the ActionHandler code, looks like this will return null if the pattern doesn't match (and maybe if there are other errors). Can you check if KitUtils.isEmpty(result) here, and only call `onResult` if it isn't empty?